app/src/main/java/space/taran/arknavigator/utils/DataStoreManager.kt

```diff
@@ -0,0 +1,36 @@
+package space.taran.arknavigator.utils
+
+import android.content.Context
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.first
+import javax.inject.Inject
+
+const val SHARED_PREFERENCES_KEY = "user_preferences"
+
+private val Context.preferencesDatastore by preferencesDataStore((SHARED_PREFERENCES_KEY))
+
+class DataStoreManager @Inject constructor(val context: Context){
```

Comment:
Can you rename to `UserPreferences` for clarity and move to mvp/model?

Reply:
The reason behind naming it DatastoreManager was to have a distinct understanding, that it's indeed Datastore, and not a SharedPreferences helper class etc.  I'll rename it to UserPreferences and move it.